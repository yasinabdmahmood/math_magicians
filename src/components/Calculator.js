import React, { useState } from 'react';
import Item from './Item';
import calculate from '../logic/calculate';

function Calculator() {
  const [obj, setObj] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const [display, setDisplay] = useState('');

  const handleObjectChange = (btn) => {
    const newObj = calculate(obj, btn);
    setObj(newObj);

    if (btn === '=') {
      setDisplay(newObj.total);
    } else if (btn === 'AC') {
      setDisplay('');
    } else {
      setDisplay(display + btn);
    }
  };

  return (
    <div className="calculator-container" data-testid="calculator-page">
      <h2>Lets do some math</h2>
      <div className="calculator">
        <div className="dark-gray" data-testid="display">{display}</div>
        <Item label="AC" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="+/-" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="%" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="รท" cl="orange" handleObjectChange={handleObjectChange} />
        <Item label="7" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="8" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="9" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="x" cl="orange" handleObjectChange={handleObjectChange} />
        <Item label="4" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="5" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="6" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="-" cl="orange" handleObjectChange={handleObjectChange} />
        <Item label="1" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="2" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="3" cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="+" cl="orange" handleObjectChange={handleObjectChange} />
        <Item label="0" cl="double light-gray" handleObjectChange={handleObjectChange} />
        <Item label="." cl="light-gray" handleObjectChange={handleObjectChange} />
        <Item label="=" cl="orange" handleObjectChange={handleObjectChange} />

      </div>
    </div>
  );
}

export default Calculator;
