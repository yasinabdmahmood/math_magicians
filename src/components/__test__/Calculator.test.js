import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from "../Calculator"

it ("Check if the component has changed", () => {
    const tree = renderer.create(
        <Calculator />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('should get 6  for (1+2)*2', () => {
    render(
        <Calculator />
    );
    const display = screen.getByTestId("display");
     const one = screen.getByText("1");
     const two = screen.getByText("2");
     const plus = screen.getByText("+");
     const multiply = screen.getByText("x");
     const equal = screen.getByText("=");
     fireEvent.click(one)
     fireEvent.click(plus)
     fireEvent.click(two)
     fireEvent.click(multiply)
     fireEvent.click(two)
     fireEvent.click(equal)
    expect(display).toHaveTextContent("6");
   
});

