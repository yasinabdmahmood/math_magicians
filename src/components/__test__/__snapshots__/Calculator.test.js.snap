// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component  1`] = `
<div
  className="calculator-container"
>
  <h2>
    Lets do some math
  </h2>
  <div
    className="calculator"
  >
    <div
      className="dark-gray"
      data-testid="display"
    />
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      AC
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      +/-
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      %
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      รท
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      7
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      8
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      9
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      x
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      4
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      5
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      6
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      -
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      1
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      2
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      3
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      +
    </button>
    <button
      className="item   double light-gray"
      onClick={[Function]}
      type="submit"
    >
      0
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      .
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Check if the component has chan 1`] = `
<div
  className="calculator-container"
>
  <h2>
    Lets do some math
  </h2>
  <div
    className="calculator"
  >
    <div
      className="dark-gray"
      data-testid="display"
    />
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      AC
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      +/-
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      %
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      รท
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      7
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      8
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      9
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      x
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      4
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      5
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      6
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      -
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      1
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      2
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      3
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      +
    </button>
    <button
      className="item   double light-gray"
      onClick={[Function]}
      type="submit"
    >
      0
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      .
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Check if the component has changed 1`] = `
<div
  className="calculator-container"
>
  <h2>
    Lets do some math
  </h2>
  <div
    className="calculator"
  >
    <div
      className="dark-gray"
      data-testid="display"
    />
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      AC
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      +/-
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      %
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      รท
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      7
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      8
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      9
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      x
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      4
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      5
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      6
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      -
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      1
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      2
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      3
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      +
    </button>
    <button
      className="item   double light-gray"
      onClick={[Function]}
      type="submit"
    >
      0
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      .
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;

exports[`renders when there are no items 1`] = `
<div
  className="calculator-container"
>
  <h2>
    Lets do some math
  </h2>
  <div
    className="calculator"
  >
    <div
      className="dark-gray"
      data-testid="display"
    />
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      AC
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      +/-
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      %
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      รท
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      7
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      8
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      9
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      x
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      4
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      5
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      6
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      -
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      1
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      2
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      3
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      +
    </button>
    <button
      className="item   double light-gray"
      onClick={[Function]}
      type="submit"
    >
      0
    </button>
    <button
      className="item   light-gray"
      onClick={[Function]}
      type="submit"
    >
      .
    </button>
    <button
      className="item   orange"
      onClick={[Function]}
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;
